### PR TITLE
feat(dli/connection): add new resource to manage privileges

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support a new resource to grant the queue binding permission to the enhance connection.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource to manage privileges.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDatasourceConnectionPrivilege_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDatasourceConnectionPrivilege_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceConnectionPrivilege_basic
=== PAUSE TestAccDatasourceConnectionPrivilege_basic
=== CONT  TestAccDatasourceConnectionPrivilege_basic
--- PASS: TestAccDatasourceConnectionPrivilege_basic (62.24s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       62.282s
```

The 404 error return when the related connection has been deleted
connection_not_found

The 404 error return when the privileges has been deleted
privilege_not_found

### CODE REVIEW

File: huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection_privilege.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+func buildModifyDatasourceConnectionPrivilegesBodyParams(d *schema.ResourceData, action string) map[string]interface{} {
+	// Default values of the privileges parameter is 'BIND_QUEUE'.
+	privileges := []string{"BIND_QUEUE"}
+	if d.Get("privileges").(*schema.Set).Len() > 0 {
+		privileges = utils.ExpandToStringListBySet(d.Get("privileges").(*schema.Set))
+	}
+
+	return map[string]interface{}{
+		"projectId": utils.ValueIngoreEmpty(d.Get("project_id")),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection_privilege.go]: project_id is required, it will never be empty

Comment (Lance52259) [reply huaweicloud/services/dli/resource_huaweicloud_dli_datasource_connection_privilege.go]: fixed

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDatasourceConnectionPrivilege_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDatasourceConnectionPrivilege_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceConnectionPrivilege_basic
=== PAUSE TestAccDatasourceConnectionPrivilege_basic
=== CONT  TestAccDatasourceConnectionPrivilege_basic
--- PASS: TestAccDatasourceConnectionPrivilege_basic (63.21s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       63.245s
```

**Pull request merged**